theme-utils.mjs

```diff
@@ -1542,33 +1542,270 @@ async function validateThemes( themes, { format, color, tableWidth } ) {
+				if (
+					value &&
+					! /^https:\/\/wordpress\.com\/themes\/[a-z0-9-]+\/?$/.test(
+						value
+					)
+				) {
+					return {
+						actual: `${ chalkStr.green(
+							attr
+						) }: ${ chalkStr.yellow( value ) }`,
+						expected: `https://wordpress.com/themes/${ chalkStr.yellow( '{slug}' ) }/`,
+						message: `${ value } is not a valid WordPress.com theme URI`,
+					};
+				}
+			},
+			isA8CAuthor: ( attr, value ) => {
+				if ( value && ! /^Automattic$/.test( value ) ) {
+					return {
+						actual: `${ chalkStr.green(
+							attr
+						) }: ${ chalkStr.yellow( value ) }`,
+						expected: `Automattic`,
+						message: `${ value } is not a valid author`,
+					};
+				}
+			},
+			isA8CAuthorUri: ( attr, value ) => {
+				if (
+					value &&
+					! /^https:\/\/automattic\.com\/$/.test( value )
+				) {
+					return {
+						actual: `${ chalkStr.green(
+							attr
+						) }: ${ chalkStr.yellow( value ) }`,
+						expected: `https://automattic.com/`,
+						message: `${ value } is not a valid Automattic author URI`,
+					};
+				}
+			},
+		};
 
-		if ( ! isSupportedWpVersion ) {
-			problems.push(
-				createProblem( {
-					type: 'warning',
-					file: styleCssPath,
-					// prettier-ignore
-					data: {
-						actual: chalkStr.yellow( wpVersion ),
-						expected: `${ chalkStr.yellow( '5.9' ) } or greater`,
-						message: `the ${ chalkStr.green( "'Requires at least'" ) } version does not support theme.json`,
+		// validate style.css metadata
+		// Spec: https://developer.wordpress.org/themes/basics/main-stylesheet-style-css/
+		const styleCssMetadata = [
+			{ attribute: 'Theme Name', required: true },
+			{
+				attribute: 'Theme URI',
+				validators: [
+					{
+						validate: validators.isUri,
+						type: 'warning',
 					},
-				} )
-			);
-		}
+					{
+						validate: validators.isA8CThemeUri,
+						type: 'warning',
+					},
+				],
+			},
+			{
+				attribute: 'Author',
+				required: true,
+				validators: [
+					{
+						validate: validators.isA8CAuthor,
+						type: 'warning',
+					},
+				],
+			},
+			{
+				attribute: 'Author URI',
+				validators: [
+					{
+						validate: validators.isUri,
+						type: 'warning',
+					},
+					{
+						validate: validators.isA8CAuthorUri,
+						type: 'warning',
+					},
+				],
+			},
+			{ attribute: 'Description', required: true },
+			{
+				attribute: 'Version',
+				required: true,
+				validators: [
+					{
+						validate: ( attr, value ) =>
+							validators.isValidVersion( attr, value, [ 3 ] ),
+						type: 'error',
+					},
+				],
+			},
+			{
+				attribute: 'Requires at least',
+				required: true,
+				validators: [
+					{
+						validate: ( attr, value ) =>
+							validators.isValidVersion( attr, value, [ 2 ] ),
+						type: 'error',
+					},
+					{
+						validate: ( attr, value ) =>
+							validators.isGreaterOrEqual(
+								attr,
+								`${ value }.0`,
+								'5.9.0'
+							),
+						type: 'error',
+					},
```

Comment:
I don't think I would add this to the main list of validators here.

I should have probably named `isSupportedWpVersion` something like `hasOfficialThemeJsonSchema` instead and had a better warning message to be more clear about that.

`5.9` is just the version where the official theme.json JSON schema was introduced.

Previously the goal was only validating theme.json files, so that's why the warning existed at the top level. Now it probably makes more sense as a warning for block themes only.

We can still validate that all Automattic/themes are block themes with a theme.json, but I'd rather do that with a `fs.existsSync( 'theme.json' )` check instead.

Reply:
Makes sense. Reverted to previous validation with additional check of `theme.json` existence.